File: ACM/Acwing/2.py
'''
Author: NEFU AB-IN
Date: 2022-03-05 16:18:39
FilePath: \Acwing\2.py
LastEditTime: 2022-07-23 14:09:05
'''

N = 1010
w, v, dp = [0] * N, [0] * N, [0] * N

n, m = map(int, input().split())
for i in range(n):
    v[i], w[i] = map(int, input().split())

for i in range(n):
    for j in range(m, v[i] - 1, -1):
        dp[j] = max(dp[j], dp[j - v[i]] + w[i])

print(dp[m])

File: GoTest/2/2.go
/*
 * @Author: NEFU AB-IN
 * @Date: 2023-08-06 23:17:06
 * @FilePath: \GoTest\2\2.go
 * @LastEditTime: 2023-08-22 11:42:20
 */
package main

import (
	"encoding/json"
	"fmt"
)

type Student struct {
	Name string `json:"name"`
	Sid  string `json:"sid"`
}

func main() {
	s := Student{Name: "jonh", Sid: "10323"}
	//序列化
	p, err := json.Marshal(s)
	if err != nil {
		panic(err)
	}
	fmt.Println(p)

	//反序列化
	err = json.Unmarshal(p, &s)
	if err != nil {
		panic(err)
	}
	fmt.Printf("%#v\n", s)
}
